### PR TITLE
Large microphone cleanup

### PR DESCRIPTION
**Pull request by dhdaines**
The microphone service code had a lot of unused and redundant stuff in it.  This makes it much simpler and also simplifies the upload code that uses it, improving error handling slightly.

### CODE REVIEW

File: packages/studio-web/src/app/microphone.service.ts [hunk 1]
```diff
@@ -1,132 +1,64 @@
...
-    return MicrophoneService.guc();
-  }
-
-  startRecording() {
-    if (this._recorderState === RecorderState.RECORDING) {
-      this.recorderError.emit(ErrorCase.ALREADY_RECORDING);
-    }
-    if (this._recorderState === RecorderState.PAUSED) {
+  async startRecording() {
+    if (this.recorder !== null && this.recorder.state == "paused") {
```
Comment (joanise) [inline packages/studio-web/src/app/microphone.service.ts:12]: `=== "paused"`?

**Review by joanise — APPROVED** [review]
Wow, that's nice. So much simpler with async that tracking the states. Thanks for that change!

File: packages/studio-web/src/app/microphone.service.ts (hunk 1 above)
Comment (dhdaines) [reply packages/studio-web/src/app/microphone.service.ts:12]: Hi! The states are defined here https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder/state (which is why it didn't make much sense to replicate them in the service ...)

**Comment by dhdaines** [comment]
> Wow, that's nice. So much simpler with async that tracking the states. Thanks for that change!

Thanks, my Mac is supposed to arrive today, so I will make sure everything works on Safari before merging

File: packages/studio-web/src/app/microphone.service.ts (hunk 1 above)
Comment (joanise) [reply packages/studio-web/src/app/microphone.service.ts:12]: Sorry, I finally realized my question was unclear. I was asking about `==` vs `===` and I'm thinking we want three equal signs here.

**Pull request merged**